googleapiclient/discovery.py

```diff
@@ -201,10 +201,8 @@ def build(serviceName,
   if resp.status >= 400:
     raise HttpError(resp, content, uri=requested_url)
 
-  try:
+  if isinstance(content, bytes):
     content = content.decode('utf-8')
-  except AttributeError:
-    pass
```

Comment:
I agree with @methane here.  Although the flexibility of handling both is nice (and how I initially handled it), httplib2 states:
> In httplib2, the response headers are strings, but the content is bytes.

From some of the docstrings in this library, 
> http: httplib2.Http, An instance of httplib2.Http or something that acts like it that HTTP requests will be made through.

So if another http client is used here, it should follow the same rules.  We should also not let those rules be broken here as well.